kernel/src/table_changes/scan_file.rs

```diff
@@ -0,0 +1,450 @@
+                    timestamp,
+                };
+                (self.callback)(&mut self.context, scan_file)
+            } else if let Some(path) = getters[7].get_opt(row_index, "scanFile.remove.path")? {
+                let deletion_vector = visit_deletion_vector_at(row_index, &getters[8..=12])?;
+                let dv_info = DvInfo { deletion_vector };
+                let partition_values = getters[13].get(
+                    row_index,
+                    "scanFile.remove.fileConstantValues.partitionValues",
+                )?;
+                let scan_file = ScanFile {
+                    tpe: ScanFileType::Remove,
+                    path,
+                    dv_info,
+                    partition_values,
+                    commit_version,
+                    timestamp,
+                };
+                (self.callback)(&mut self.context, scan_file)
+            } else if let Some(path) = getters[14].get_opt(row_index, "scanFile.cdc.path")? {
+                let partition_values = getters[15]
+                    .get(row_index, "scanFile.cdc.fileConstantValues.partitionValues")?;
+                let scan_file = ScanFile {
+                    tpe: ScanFileType::Cdc,
+                    path,
+                    dv_info: DvInfo {
+                        deletion_vector: None,
+                    },
+                    partition_values,
+                    commit_version,
+                    timestamp,
+                };
+
+                (self.callback)(&mut self.context, scan_file)
+            }
+        }
+        Ok(())
+    }
+
+    fn selected_column_names_and_types(
+        &self,
+    ) -> (
+        &'static [crate::expressions::ColumnName],
+        &'static [DataType],
+    ) {
+        static NAMES_AND_TYPES: LazyLock<ColumnNamesAndTypes> =
+            LazyLock::new(|| scan_row_schema().leaves(None));
+        NAMES_AND_TYPES.as_ref()
+    }
+}
+
+/// Get the schema that scan rows (from [`TableChanges::scan_data`]) will be returned with.
+///
+/// It is:
+/// ```ignored
+/// {
+///   add: {
+///     path: string,
+///     deletionVector: {
+///       storageType: string,
+///       pathOrInlineDv: string,
+///       offset: int,
+///       sizeInBytes: int,
+///       cardinality: long,
+///     },
+///     fileConstantValues: {
+///       partitionValues: map<string, string>
+///     }
+///   }
+///   remove: {
+///     path: string,
+///     deletionVector: {
+///       storageType: string,
+///       pathOrInlineDv: string,
+///       offset: int,
+///       sizeInBytes: int,
+///       cardinality: long,
+///     },
+///     fileConstantValues: {
+///       partitionValues: map<string, string>
+///     },
+///   }
+///   cdc: {
+///     path: string,
+///     fileConstantValues: {
+///       partitionValues: map<string, string>
+///     }
+///   }
+///   commit_timestamp: long,
+///   commit_version: long
+///
+/// }
+/// ```
+pub(crate) fn scan_row_schema() -> SchemaRef {
```

Comment:
probably doesn't need to be `pub(crate)`. I think it should only be used in this file

Reply:
This will be used by the `LogReplayScanner` since this is the output schema for the transformed EngineData.